formats/cbor/commonMain/src/kotlinx/serialization/cbor/internal/Streams.kt

```diff
@@ -43,12 +43,18 @@ internal class ByteArrayOutput {
     private var position: Int = 0
 
     private fun ensureCapacity(elementsToAppend: Int) {
-        if (position + elementsToAppend <= array.size) {
+        val requiredCapacityLong = position.toLong() + elementsToAppend.toLong()
+        if (requiredCapacityLong > Int.MAX_VALUE) {
+            throw IllegalArgumentException("Required capacity exceeds maximum array size (Int.MAX_VALUE).")
```

Comment:
```suggestion
            throw SerializationException("Capacity required to hold a value serialized to CBOR ($requiredCapacityLong) exceeds maximum supported byte array length (${Int.MAX_VALUE}).")
```

It's better to:
- show the exact value that triggered an error (it might be worth to include the exact `elementsToAppend` value to the error message too, as it might help to debug an issue, when someone will have to)
- be more precise about what's happening, as it may help users facing the issue to deal with it
- throw `SerializationException` to depict that it's a problem with serializing some particular object, and it's not some internal assertion being failed.